### ISSUE TITLE
Build report tab seems to not become visible in multi-stage YAML

### ISSUE THREAD

**Issue by Mobrockers** [stryker-mutator/azure-devops-mutationreport-publisher#32]

**Comment by robertlyson** [comment]
@Mobrockers any suggestions on where I need to look to prepare a fix?

**Comment by Mobrockers** [comment]
@robertlyson the most likely places would be either for yaml pipelines there is a different contribution extension point than for legacy builds, or the supportstask works differently for yaml pipelines.

**Comment by robertlyson** [comment]
Found a few related things:
- protractor solution for the issue 
https://github.com/maciejmaciejewski/azure-pipelines-protractor/pull/11
https://github.com/maciejmaciejewski/azure-pipelines-protractor/blob/master/azure-devops-extension.json
- apparently official sample has the same issue
https://github.com/microsoft/vsts-extension-samples/issues/110
- and looks like docs have an issue with documenting the behavior :)
https://github.com/MicrosoftDocs/azure-devops-docs/issues/7728

**Comment by Mobrockers** [comment]
Awesome finds :)

I am especially interested in that dynamic keyword used by protractor and what it does. They also have support for multiple reports which is something we also need, so that should help very much in figuring out how to get this working!

Are you interested in giving it a shot or do you think that's a bit much as a first contribution?

If you are interested, this repo should be mostly setup to be able to live debug the web extension I think but I haven't tested it. This blog post describes how it should be set up to be able to run the web extension locally without uploading to azure devops: https://github.com/microsoft/azure-devops-extension-hot-reload-and-debug

Then, if you want to try out the extension in an actual azure devops, you're gonna have to set up a publisher account at https://marketplace.visualstudio.com/manage, upload the extension as private (using a different extension id) and share the extension with your own private azure devops somewhere.

**Comment by robertlyson** [comment]
@Mobrockers thanks for the resources, that will make my contribution easier, will give it a try over the weekend :)

**Comment by Mobrockers** [comment]
If you have any questions I should be available this weekend. You can contact me here or on our slack.

**Comment by robertlyson** [comment]
@Mobrockers I was only able to publish a test extension package but couldn't debug it locally yet. If you like you can give the issue try, if not I'm happy to continue my investigation over the next weekend. I just wanted to share my update.

**Comment by Mobrockers** [comment]
Thanks for the update!

**Comment by Mobrockers** [comment]
Hi @robertlyson 

I think the biggest issue was actually in the deployment itself. I had set updateTasksId: true, which generates the task id's based on the publisher and extension name. I had not used this when testing, but added it as part of the public release when I added a proper release pipeline for the extension..

Turns out when you change task id's in the pipeline, and those task id's are used for the 'contributingtask' section, those are not changed.. :/ Which means there was a mismatch and the tab never became visible. Which also means this issue was probably not only with YAML pipelines but with all pipelines... 🤦‍♀️ 

I'm soon releasing a new version which should fix this problem.
